manifests/config.pp

```diff
@@ -39,4 +50,59 @@
     group   => $otelcol::config_file_group,
     mode    => $otelcol::config_file_mode,
   }
+
+  if($otelcol::receivers) {
+    $otelcol::receivers.each|String $rname, Hash $rvalue| {
```

Comment:
What we often do is make a parameter something like `Array $receivers = []` so you can safely refactor it to always iterate:
```suggestion
  $otelcol::receivers.each|String $rname, Hash $rvalue| {
```